sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -55,7 +55,7 @@ trait DataSourceScanExec extends LeafExecNode {
   // Metadata that describes more details of this scan.
   protected def metadata: Map[String, String]
 
-  protected val maxMetadataValueLength = 100
+  protected val maxMetadataValueLength = conf.maxMetadataValueLength
```

Comment:
Have you checked the related PR? https://github.com/apache/spark/pull/29688/files#diff-2a91a9a59953aa82fa132aaf45bd731bR58

Reply:
> Have you checked the related PR? https://github.com/apache/spark/pull/29688/files#diff-2a91a9a59953aa82fa132aaf45bd731bR58

Sorry, don't notice that pr. I have similar ideal like that pr too.  I'm going to restore this change and deal with the two PR conflicts  finally.

Reply:
link: https://github.com/apache/spark/pull/29688